src/sfizz/Opcode.cpp

```diff
@@ -117,6 +117,116 @@ OpcodeCategory Opcode::identifyCategory(absl::string_view name)
     return category;
 }
 
+template <typename T>
+absl::optional<T> readInt_(OpcodeSpec<T> spec, absl::string_view v)
+{
+    size_t numberEnd = 0;
+
+    if (numberEnd < v.size() && (v[numberEnd] == '+' || v[numberEnd] == '-'))
+        ++numberEnd;
+
+    while (numberEnd < v.size() && absl::ascii_isdigit(v[numberEnd]))
+        ++numberEnd;
+
+    if (numberEnd == 0 && (spec.flags & kCanBeNote))
+        return readNoteValue(v);
+
+    v = v.substr(0, numberEnd);
+
+    int64_t returnedValue;
+    if (!absl::SimpleAtoi(v, &returnedValue))
+        return absl::nullopt;
+
+    if (returnedValue > static_cast<int64_t>(spec.bounds.getEnd())) {
+        if (spec.flags & kEnforceUpperBound)
+            return spec.bounds.getEnd();
+
+        if (spec.flags & kIgnoreOOB)
+            return {};
+    }
+
+    if (returnedValue < static_cast<int64_t>(spec.bounds.getStart())) {
+        if (spec.flags & kEnforceLowerBound)
+            return spec.bounds.getStart();
+
+        if (spec.flags & kIgnoreOOB)
+            return {};
+    }
+
+    T castValue = static_cast<T>(returnedValue);
+    if ((castValue != returnedValue) & kIgnoreOOB)
```

Comment:
This conditional is looking suspicious.
```suggestion
    if (castValue != returnedValue && (spec.flags & kIgnoreOOB))
```

I'm not sure what it's trying to accomplish though.
do we have to repair values in case of an overflow?

Reply:
Yes it's wrong. I'd say in case of an overflow we should probably ignore the input.